cmd/kpod/kill.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/kubernetes-incubator/cri-o/libkpod"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+	"os"
+)
+
+var (
+	killFlags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "signal, s",
+			Usage: "Signal to send to the container",
+			Value: "KILL",
+		},
+	}
+	killDescription = "Kill one or more running containers"
+	killCommand     = cli.Command{
+		Name:        "kill",
+		Usage:       fmt.Sprintf("Kill one or more running containers with a specific signal"),
+		Description: killDescription,
+		Flags:       killFlags,
+		Action:      killCmd,
+		ArgsUsage:   "",
+	}
+)
+
+// killCmd kills one or more containers with a signal
+func killCmd(c *cli.Context) error {
+	args := c.Args()
+	if len(args) == 0 {
+		return errors.Errorf("specify one or more containers to kill")
+	}
+	config, err := getConfig(c)
+	if err != nil {
+		return errors.Wrapf(err, "could not get config")
+	}
+	server, err := libkpod.New(config)
+	if err != nil {
+		return errors.Wrapf(err, "could not get container server")
+	}
+	defer server.Shutdown()
+	err = server.Update()
+	if err != nil {
+		return errors.Wrapf(err, "could not update list of containers")
+	}
+	hadError := false
+	killSignal := c.String("signal")
+	for _, container := range c.Args() {
+		id, err := server.ContainerKill(container, killSignal)
+		if err != nil {
+			hadError = true
+			fmt.Println(err)
+		} else {
+			fmt.Println(id)
+		}
+	}
+	if hadError {
+		os.Exit(1)
```

Comment:
Not sure this is the correct way to do this, this will cause stuff not to be cleaned up, I believe.  I think we need to save the last error and return it.

Something like 
```
lastError=nil
killSignal := c.String("signal")
for _, container := range c.Args() {
id, err := server.ContainerKill(container, killSignal)
if err != nil {
               If lastError != nil {
			fmt.Println(lastError)
               }
               lastError = errors.Wrap(err, "unable to kill %v", container)
		} else {
			fmt.Println(id)
		}
	}

return lastError
```

Reply:
What if there are multiple errors?  Seems kind of odd that way too.

And this way, won't the lastError be printed twice then?

Reply:
No we are only printing the last error once. since we will not print the last error on the first error.  When we get the second error, we print the first, when we get the third print the second ...
Then we always return the "lastError"

Reply:
Will implement

Reply:
For simplicity's sake - can we just print every error that happens, and return a generic "error killing some containers" error instead of os.Exit(1)?

Reply:
I would prefer to not print out two error messages, which is why I prefer mine.


```
kpod kill foobar
Foobar container does not exist
error killing some containers
```

Vs

```
kpod kill foobar
Foobar container does not exist
```

Reply:
I would remove the errors.Wrapf in your example, then - otherwise the last error will be printed differently than the others

Reply:
Well it should always print the lastError, which would be a wrapped error.  I don't think it would be any different but @baude can do a simple test to see what it looks like


kpod kill foo bar

And see if it prints the message twice with the same syntax.

unable to kill "foo": Container does not exist
unable to kill "bar": Container does not exist

Reply:
Should also check to make sure they're going to the same stream. I think the CLI package prints errors returned to stderror instead of stdout?